crates/oxc_regular_expression/src/visit.rs

```diff
@@ -0,0 +1,391 @@
+#![allow(unused_variables, clippy::wildcard_imports)]
+use walk::walk_pattern;
+
+use crate::ast::{
+    Alternative, BoundaryAssertion, CapturingGroup, Character, CharacterClass,
+    CharacterClassContents, CharacterClassEscape, CharacterClassRange, ClassString,
+    ClassStringDisjunction, Disjunction, Dot, IgnoreGroup, IndexedReference, LookAroundAssertion,
+    NamedReference, Pattern, Quantifier, Term, UnicodePropertyEscape,
+};
+use walk::*;
+
+#[derive(Copy, Clone, Debug)]
+pub enum RegexAstKind<'a> {
```

Comment:
IMO: I feel it might be better to avoid using the term `Regex`. It brings to mind the `regex` crate. 👀 

Although it's a bit verbose, how about `RegularExpressionAstKind`?

Reply:
hm, I'm not strongly opposed to `RegularExpressionAstKind`, but it is kind of long. maybe to differentiate from the `regex` crate, we could use the term `RegExp` instead? so it would be `RegExpAstKind` instead. That is more similar to how the ECMA262 spec refers to it.

Reply:
Personally, I still prefer `RegularExpressionAstKind` (since it's the `AstKind` of `oxc_regular_expression`), but I think `RegExpAstKind` would also be fine. 👍🏻 

---

BTW, (feel free to ignore this if I'm wrong, 😓) is it possible to alias an export with the same name in Rust?

```rs
use oxc_ast::AstKind;
use oxc_regular_expression::AstKind as RegExpAstKind; // 👈🏻 
```

If it's possible, I’m starting to think that simply leaving it as `AstKind` might be a good option.

Reply:
I think that's definitely possible, though it's less ideal for the linter codebase since it's pretty common to import AstKind too. Let's stick with this name for now and we can always change it later?

Reply:
> that's definitely possible

Thanks, I learned something new. 😃 

> Let's stick with this name for now

🙆🏻